vllm/v1/core/single_type_kv_cache_manager.py

```diff
@@ -281,11 +280,17 @@ def remove_skipped_blocks(self, request_id: str,
         pass
 
     def get_num_common_prefix_blocks(self, request_id: str,
-                                     num_running_requests: int) -> int:
+                                     running_request_ids: set[str]) -> int:
         blocks = self.req_to_blocks[request_id]
         num_common_blocks = 0
+        num_running_requests = len(running_request_ids)
+
         for block in blocks:
-            if block.ref_cnt == num_running_requests:
+            actual_running_refs = sum(1 for req_id in running_request_ids
+                                      if req_id in self.req_to_blocks
+                                      and block in self.req_to_blocks[req_id])
```

Comment:
`block in self.req_to_blocks[req_id]`  is an O(num_block) operation, making this function pretty slow.

Reply:
@heheda12345 thanks for the coments I am looking into it, but can you also please suggest something to make it optimal?

Reply:
I'm thinking of
```
    def get_num_common_prefix_blocks(self, running_request_id: str,
                                     num_running_requests: int, transfering_request_ids: list[str]) -> int:
        running_blocks = self.req_to_blocks[running_request_id]
        transfering_blocks = [self.req_to_blocks[req_id] for req_id in transfering_request_ids]
        num_common_blocks = 0
        for i, block in enumerate(running_blocks):
            num_transfering_blocks = sum(1 for blocks in transfering_blocks if blocks[i].block_id == running_blocks[i].block_id)
            if block.ref_cnt == num_running_requests + num_transfering_blocks:
                num_common_blocks += 1
            else:
                break
```